### ISSUE TITLE
Add compile options to libspdm_gen_x509_csr

### ISSUE THREAD

**Issue by rw8896** [DMTF/libspdm#2825]
https://github.com/DMTF/libspdm/blob/b327ead227f14998789bca7d045af17b99dd6d50/os_stub/cryptlib_mbedtls/pk/x509.c#L1993
https://github.com/DMTF/libspdm/blob/b327ead227f14998789bca7d045af17b99dd6d50/os_stub/cryptlib_openssl/pk/x509.c#L2495

The implementation could use LIBSPDM_ENABLE_CAPABILITY_CSR_CAP to wrap the whole function.
And it should add crypto options (e.g. LIBSPDM_RSA_SSA_2048_SUPPORT) to wrap around the crypto operation code.

**Comment by steven-bellock** [comment]
@rw8896 is this to save on code size?

**Comment by rw8896** [comment]
Not really. It caused compiler errors as RSA was not enabled in my mbedtls configuration.

After looking into this function further, I think it doesn't allow the caller to specify the pathLen in basic constraints.
If that's the case, maybe pathLen should be added as a input to make this API more general?

**Comment by jyao1** [comment]
@rw8896 , I am not sure if I understand your problem statement.

Usually, it is more helpful to describe what problem you have meet, than just describe the solution.

May I know what compiler error you have meet?

Based on my understand, **if you customize the mbedtls, then you can also customize the cryptolib_mbedtls.**

**Comment by rw8896** [comment]
The following code calls mbedtls_pk_rsa but mbedtls_pk_rsa is not declared/implemented when MBEDTLS_RSA_C is not defined.

https://github.com/DMTF/libspdm/blob/b327ead227f14998789bca7d045af17b99dd6d50/os_stub/cryptlib_mbedtls/pk/x509.c#L2034-L2055

**Comment by jyao1** [comment]
I still not get it. Why not just use LIBSPDM_ENABLE_CAPABILITY_CSR_CAP ? or just return false, if the crypto lib does not support.

**Comment by rw8896** [comment]
Hi Jiewen,

We encountered build errors because LIBSPDM_ENABLE_CAPABILITY_CSR_CAP is enabled in our project, but we use our own functions to populate the CSR to libspdm. Additionally, the mbedtls on our platform does not support RSA.

Initially, I considered adding RSA wrapping to this function. However, upon further consideration, I believe it is the integrator's responsibility to generate the CSR, and libspdm may not need to provide libspdm_gen_x509_csr.

Three potential options to resolve this:

1.  Add algorithm support compile options in libspdm_gen_x509_csr.
2.  Add LIBSPDM_CSR_GEN_SUPPORT to exclude this function.
3.  Remove this function entirely, as the integrator can handle CSR generation when implementing the device secret library.
